src/main/java/me/jellysquid/mods/sodium/common/util/PathUtil.java

```diff
@@ -0,0 +1,21 @@
+package me.jellysquid.mods.sodium.common.util;
+
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class PathUtil {
+    /**
+     * Resolves a timestamped sibling of the base path - {@code "{base_dir}/{base_name}_{tag}_{timestamp}.{base_ext}"}.
+     * @param base Base path
+     * @param tag Tag to add to sibling's name
+     * @return Sibling path
+     */
+    public static Path resolveTimestampedSibling(Path base, String tag) {
+        LocalDateTime now = LocalDateTime.now();
+        String backupName = base.getFileName().toString();
+        backupName = backupName.substring(0, backupName.lastIndexOf('.') - 1);
+        backupName += "_" + tag + "_" + now.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME) + ".json";
```

Comment:
This causes issues with path resolution because `:` is not a valid file name. Reported on the Discord here: https://discord.com/channels/602796788608401408/778685768041627719/819478369418215424

Reply:
great job, past me. what do I even pay you for

Reply:
Should be fixed in https://github.com/CaffeineMC/sodium-fabric/pull/571/commits/220c417fca9ae23a142661874879a31ee300cab5.